gobblin-compaction/src/main/java/gobblin/compaction/conditions/RecompactionCondition.java

```diff
@@ -0,0 +1,9 @@
+package gobblin.compaction.conditions;
+import gobblin.compaction.dataset.DatasetMetric;
+
+/**
+ * RecompactionCondition interface.
```

Comment:
Please add more java doc. What is abstraction indented for? Where can this be used